### ISSUE TITLE
Confusion with require-input-label

### ISSUE THREAD

**Issue by locks** [ember-template-lint/ember-template-lint#1835]
I have the following template code that triggered the `require-input-label` lint:

```
<div class="search-input">
  <label class="visually-hidden" for="search-input">Search</label>
  <input
    id="search-input"
    type="search"
    placeholder="Search deprecations"
    autocomplete="off"
    aria-label="Search"
    {{on "input" (perform this.searchService.search value="target.value")}}
  >
</div>
```

I was a bit confused as I have `label` and an `aria-label`, and wasn't sure how to address it.
I ended up removing `aria-label` at the suggestion of someone in Discord.

**Comment by lifeart** [comment]
@MelSumner should we count is as "conflict"? (input already have label reference and aria-label is redundant)

**Comment by MelSumner** [comment]
@locks did you look at the documentation for the rule? If so, did you find the documentation sufficient to help you understand what was going on? 

When @rwjblue and I were discussing what to put in the error feedback, we decided the goal was to stay terse, which should encourage the developer to read the rule documentation and thereby learn more about the issue.

**Comment by runspired** [comment]
I think there *may* be a bug here but I'm not sure my knowledge of a11y isn't just limited. The following pattern will trigger this rule (@jenweber and I discovered this pairing on an upgrade to the ember-api-docs app). As far as I can tell, this isn't a violation, 

```hbs
 <label class="access-checkbox">
          <input id="inherited-toggle"
                 data-test-checkbox="inherited"
                 type="checkbox"
                 checked={{filterData.showInherited}}
                 onchange={{action "updateFilter" "showInherited"}}>
          Inherited
        </label>
```

The issue appears to be that the input is *both* nested in a label *and* has an `id`. Removing the ID removes the lint error. Is having an id on an input inside of a label invalid?

**Comment by MelSumner** [comment]
@runspired yeah that would be a bug in that case; it's perfectly valid to have a nested label/input and an id.

**Comment by bertdeblock** [comment]
I think this one can be closed since https://github.com/ember-template-lint/ember-template-lint/pull/1994 was merged.

**Comment by MelSumner** [comment]
Issue should be resolved with #1994 but please re-open if you still see any issues. Thank you!
